src/ex-gen/Program.cs

```diff
@@ -133,91 +143,58 @@ private static void Run(string sourcePath, string inclusionFile, string exclusio
+            var emptyMain = @"internal static class Program
+{
+    public static void Main()
+    {
+    }
+}";
 
-                var extractedFolderName = Path.GetFileNameWithoutExtension(fileName).Replace(".tar", "");
-                var exractedFolderPath = Path.Combine(tempFolder, extractedFolderName);
+            var directoryPath = Path.GetDirectoryName(projectFilePath);
+            Directory.CreateDirectory(directoryPath);
 
-                using (var fileStream = File.OpenRead(fileName))
-                using (var archive = OpenArchive(fileStream))
-                {
-                    var selectedEntries = archive.Entries.Where(
-                        e => e.Key.StartsWith(@"./shared/Microsoft.NETCore.App/") ||
-                        e.Key.StartsWith(@"shared/Microsoft.NETCore.App/") ||
-                        e.Key.StartsWith(@"shared\Microsoft.NETCore.App\"));
-                    if (!selectedEntries.Any())
-                        throw new InvalidDataException($"No archive selected to be extracted from {fileName} at {tempFolder}");
-
-                    foreach (var entry in selectedEntries)
-                        entry.WriteToDirectory(exractedFolderPath, options);
-                }
+            File.WriteAllText(projectFilePath, contents);
 
-                File.Delete(fileName);
-            }
+            var programCs = Path.Combine(directoryPath, "Program.cs");
+            File.WriteAllText(programCs, emptyMain);
         }
 
-        private static IArchive OpenArchive(FileStream fileStream)
+        private static void PublishProject(string projectFilePath, string outputPath, string rid)
         {
-            var extension = Path.GetExtension(fileStream.Name);
-            switch (extension)
-            {
-                case ".tar":
-                    return TarArchive.Open(fileStream);
-                case ".zip":
-                    return ZipArchive.Open(fileStream);
-                default:
-                    throw new NotImplementedException($"Unknown file {extension}");
-            }
+            var command = "dotnet";
+            var args = $@"publish --output ""{outputPath}"" --runtime ""{rid}""";
+            var workingDirectoryPath = Path.GetDirectoryName(projectFilePath);
+
+            var process = new Process();
+            process.StartInfo.FileName = command;
+            process.StartInfo.Arguments = args;
+            process.StartInfo.WorkingDirectory = workingDirectoryPath;
+            process.StartInfo.RedirectStandardError = true;
+            process.StartInfo.RedirectStandardOutput = true;
+            process.StartInfo.UseShellExecute = false;
+            process.OutputDataReceived += (s, d) => Console.Out.WriteLine(d.Data);
+            process.ErrorDataReceived += (s, d) => Console.Error.WriteLine(d.Data);
+            process.Start();
+            process.BeginOutputReadLine();
+            process.BeginErrorReadLine();
+            process.WaitForExit();
```

Comment:
I expect dotnet to set an error code in case of failure, checking the `ExitCode` and throwing seems good.

Reply:
Good call, fixed.